static/js/components/dashboard/FinalExamCard.js

```diff
@@ -137,14 +140,44 @@ const schedulableCourseList = R.compose(
   R.defaultTo({}),
 );
 
-const schedulableCard = (profile, program, navigateToProfile, submitPearsonSSO, pearson) => cardWrapper(
+const renderPearsonTOSDialog = (open, show, submitPearsonSSO) => (
+  <Dialog
+    key="pearson-tos-dialog"
+    contentClassName="dialog content"
+    className="dialog-to-pearson-site"
+    open={open}
+    onRequestClose={() => show(false)}
+    actions={dialogActions(() => show(false), () => submitPearsonSSO(), false, "CONTINUE")}
+    autoScrollBodyContent={true}>
+    <div className="dialog-container">
+      <img src="/static/images/pearson_vue.png" width="180"/>
+      <h3 className="dialog-title">Test Registration is completed on the <br/>Pearson VUE website</h3>
+      <p>
+        I acknowledge that by clicking Continue I will be leaving the MicroMasters <br/>website and going to
+        the Pearson VUE website, and that I accept the <br/>Pearson VUE Group’s <a target="_blank"
+          href="https://home.pearsonvue.com/Legal/Privacy-and-cookies-policy.aspx">Terms of Service</a>.
+      </p>
+    </div>
+  </Dialog>
+);
+
+const schedulableCard = (
+  profile,
+  program,
+  navigateToProfile,
+  pearson,
+  showPearsonTOSDialog,
+  open,
+  submitPearsonSSO
+) => cardWrapper(
+  renderPearsonTOSDialog(open, showPearsonTOSDialog, submitPearsonSSO),
   accountCreated(profile, navigateToProfile),
   <div key="schedulable" className="exam-scheduling">
     <SpinnerButton
       className="mdl-button exam-button"
       component={Button}
       spinning={isProcessing(pearson)}
```

Comment:
this is no longer the button that should be spinning - can we change it around so that the 'continue' button in the dialog spins while the SSO stuff is in-flight?